src/presign/round_one.rs

```diff
@@ -22,16 +22,39 @@ use serde::{Deserialize, Serialize};
 use std::fmt::Debug;
 use zeroize::ZeroizeOnDrop;
 
+use super::round_three::SecretBigNumber;
+
+#[derive(Clone, ZeroizeOnDrop)]
+pub(crate) struct SecretNonce(Nonce);
+
+impl SecretNonce {
+    pub fn get_nonce(&self) -> &Nonce {
```

Comment:
We should change the name of this method to make it explicit that you are being given direct access to the underlying secret nonce. A doc comment on this method should state the same thing and warn against cloning the value.